v2/tests/log_sink_etw_int/log_sink_etw_int.c

```diff
@@ -0,0 +1,683 @@
+    TRACEHANDLE trace_session_handle;
+
+    (void)printf("Starting trace session %s\r\n", trace_session_name);
+
+    EVENT_TRACE_PROPERTY_DATA start_event_trace_property_data = { 0 };
+
+    start_event_trace_property_data.props.Wnode.BufferSize = sizeof(start_event_trace_property_data);
+    start_event_trace_property_data.props.Wnode.Flags = WNODE_FLAG_TRACED_GUID;
+    start_event_trace_property_data.props.LogFileMode = EVENT_TRACE_REAL_TIME_MODE;
+    start_event_trace_property_data.props.MinimumBuffers = 4;
+    start_event_trace_property_data.props.MaximumBuffers = 4;
+    start_event_trace_property_data.props.BufferSize = 64 * 1024;
+    start_event_trace_property_data.props.FlushTimer = 1;
+    start_event_trace_property_data.props.LogFileNameOffset = 0;
+    start_event_trace_property_data.props.LoggerNameOffset = offsetof(EVENT_TRACE_PROPERTY_DATA, logger_name);
+
+    POOR_MANS_ASSERT(StartTraceA(&trace_session_handle, trace_session_name, &start_event_trace_property_data.props) == ERROR_SUCCESS);
+
+    (void)strcpy(logger_name, start_event_trace_property_data.logger_name);
+
+    // enable our desired provider
+    ULONG enable_trace_result = EnableTraceEx2(trace_session_handle, (LPCGUID)&provider_guid, EVENT_CONTROL_CODE_ENABLE_PROVIDER,
+        trace_level, 0, 0, 0, NULL
+    );
+    POOR_MANS_ASSERT(enable_trace_result == ERROR_SUCCESS);
+
+    return trace_session_handle;
+}
+
+static void stop_trace(TRACEHANDLE trace_session_handle)
+{
+    EVENT_TRACE_PROPERTY_DATA stop_event_trace_property_data = { 0 };
+
+    stop_event_trace_property_data.props.Wnode.BufferSize = sizeof(stop_event_trace_property_data);
+    stop_event_trace_property_data.props.Wnode.Flags = WNODE_FLAG_TRACED_GUID;
+    stop_event_trace_property_data.props.LoggerNameOffset = offsetof(EVENT_TRACE_PROPERTY_DATA, logger_name);
+    stop_event_trace_property_data.props.LogFileNameOffset = 0;
+
+    (void)printf("Stopping trace session %s\r\n", trace_session_name);
+    POOR_MANS_ASSERT(StopTraceA(trace_session_handle, NULL, &stop_event_trace_property_data.props) == ERROR_SUCCESS);
+}
+
+DWORD WINAPI process_trace_thread_func(void* arg)
+{
+    (void)arg;
+
+    // open the trace and process all events
+    ULONG process_trace_result = ProcessTrace(&trace, 1, NULL, NULL);
+    POOR_MANS_ASSERT(process_trace_result == ERROR_SUCCESS);
+
+    return 0;
+}
+
+#define TEST_PARSE_START_DELAY 5000 // ms
+#define TEST_WAITTIME_NO_EVENT 20000 // ms
+
+static void start_parse_events(void)
+{
+    EVENT_TRACE_LOGFILEA log_file = { 0 };
+
+    parsed_event_count = 0;
+
+    log_file.ProcessTraceMode = PROCESS_TRACE_MODE_REAL_TIME | PROCESS_TRACE_MODE_EVENT_RECORD;
+    log_file.EventRecordCallback = event_trace_record_callback;
+    log_file.BufferCallback = event_trace_buffer_callback;
+    log_file.LoggerName = logger_name;
+    log_file.Context = NULL;
+
+    trace = OpenTraceA(&log_file);
+    POOR_MANS_ASSERT(trace != INVALID_PROCESSTRACE_HANDLE);
```

Comment:
>trace [](http://example.com/codeflow?start=21&length=5)

looks like "trace" should be a parameter passed to process_trace_thread_func. Just saying that there's a bit too many globals here.